shipgen/EvtCalcGenerator.cxx

```diff
@@ -0,0 +1,121 @@
+  if (fTree->FindBranch("e_llp") ){fTree->SetBranchAddress("e_llp", &e_llp);}
+  if (fTree->FindBranch("pdg_llp") ){fTree->SetBranchAddress("pdg_llp", &pdg_llp);}
+  if (fTree->FindBranch("decay_prob") ){fTree->SetBranchAddress("decay_prob", &decay_prob);}
+  if (fTree->FindBranch("vx") ){fTree->SetBranchAddress("vx", &vx);}
+  if (fTree->FindBranch("vy") ){fTree->SetBranchAddress("vy", &vy);}
+  if (fTree->FindBranch("vz") ){fTree->SetBranchAddress("vz", &vz);}
+
+  if (fTree->FindBranch("px_prod1") ){fTree->SetBranchAddress("px_prod1", &px_prod1);}
+  if (fTree->FindBranch("py_prod1") ){fTree->SetBranchAddress("py_prod1", &py_prod1);}
+  if (fTree->FindBranch("pz_prod1") ){fTree->SetBranchAddress("pz_prod1", &pz_prod1);}
+  if (fTree->FindBranch("e_prod1") ){fTree->SetBranchAddress("e_prod1", &e_prod1);}
+  if (fTree->FindBranch("pdg_prod1") ){fTree->SetBranchAddress("pdg_prod1", &pdg_prod1);}
+
+  if (fTree->FindBranch("px_prod2") ){fTree->SetBranchAddress("px_prod2", &px_prod2);}
+  if (fTree->FindBranch("py_prod2") ){fTree->SetBranchAddress("py_prod2", &py_prod2);}
+  if (fTree->FindBranch("pz_prod2") ){fTree->SetBranchAddress("pz_prod2", &pz_prod2);}
+  if (fTree->FindBranch("e_prod2") ){fTree->SetBranchAddress("e_prod2", &e_prod2);}
+  if (fTree->FindBranch("pdg_prod2") ){fTree->SetBranchAddress("pdg_prod2", &pdg_prod2);}
+
+  if (fTree->FindBranch("px_prod3") ){fTree->SetBranchAddress("px_prod3", &px_prod3);}
+  if (fTree->FindBranch("py_prod3") ){fTree->SetBranchAddress("py_prod3", &py_prod3);}
+  if (fTree->FindBranch("pz_prod3") ){fTree->SetBranchAddress("pz_prod3", &pz_prod3);}
+  if (fTree->FindBranch("e_prod3") ){fTree->SetBranchAddress("e_prod3", &e_prod3);}
+  if (fTree->FindBranch("pdg_prod3") ){fTree->SetBranchAddress("pdg_prod3", &pdg_prod3);}
+
+  return kTRUE;
+}
+// -------------------------------------------------------------------------
+
+
+// -----   Destructor   ----------------------------------------------------
+EvtCalcGenerator::~EvtCalcGenerator()
+{
+ // cout << "destroy Ntuple" << endl;
+ fInputFile->Close();
+ fInputFile->Delete();
+ delete fInputFile;
+}
+// -------------------------------------------------------------------------
+
+// -----   Passing the event   ---------------------------------------------
+Bool_t EvtCalcGenerator::ReadEvent(FairPrimaryGenerator* cpg)
+{
+  if (fn==fNevents){ LOG(WARNING) << "End of input file. Rewind."; }
+  fTree->GetEntry(fn);
+  fn++;
+  if (fn %100==0)  cout << "Info EvtCalcGenerator: event nr "<< fn << endl;
+
+  //Vertex coordinates in the SHiP reference frame, expressed in [cm]
+  Double_t space_unit_conv = 100.;            // m to cm 
+  Double_t coord_shift     = (zDecayVolume - ztarget)/space_unit_conv; // units m
+  Double_t vx_transf = vx * space_unit_conv; // units cm
+  Double_t vy_transf = vy * space_unit_conv; // units cm
+  Double_t vz_transf = (vz - coord_shift) * space_unit_conv; // units cm
+
+  Double_t c = 2.99792458e+10; // speed of light [cm/s]
```

Comment:
It would be good to use the one provided by `ShipUnit.h`

Reply:
Hi @olantwin , I haven't found time to complete this task yet. It seems that something is not adding up when using the Shipunits namespace. I am happy to proceed with the pull request without this enhancement if you think it is a good idea.

Reply:
That's fine, let's for now stick to the version that works.

Could you please open an issue for this? I think we keep redefining the speed of light in many many files, so we should try to fix this throughout the code base anyway.

Reply:
It's already there: Issue #541